sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -937,7 +937,8 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
  */
 case class OptimizeCodegen(conf: CatalystConf) extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
-    case e @ CaseWhen(branches, _) if branches.size < conf.maxCaseBranchesForCodegen =>
+    case e @ CaseWhen(branches, elseBranch)
+        if branches.size + elseBranch.size <= conf.maxCaseBranchesForCodegen =>
```

Comment:
Reading the case takes a while and  and I think it'd greatly benefit from introducing a local `def` - a predicate - for the condition (I can't figure out a name for this, sorry)